src/seedu/addressbook/commands/CommandResult.java

```diff
@@ -13,17 +13,34 @@
     /** The feedback message to be shown to the user. Contains a description of the execution result */
     public final String feedbackToUser;
 
+    /** This boolean decides whether listing will go with the command */
+    public final boolean hasToList;
+
     /** The list of persons that was produced by the command */
     private final List<? extends ReadOnlyPerson> relevantPersons;
 
+    private final List<? extends ReadOnlyPerson> allPersons;
+
     public CommandResult(String feedbackToUser) {
         this.feedbackToUser = feedbackToUser;
         relevantPersons = null;
+        this.hasToList = false;
+        allPersons = null;
+    }
+
+    // This method use for commands that need auto listing
+    public CommandResult(String feedbackToUser, boolean hasToList, List<? extends ReadOnlyPerson> allPersons) {
```

Comment:
Try to put optional arguments at the end of the list. This makes the code used to call this method more consistent.